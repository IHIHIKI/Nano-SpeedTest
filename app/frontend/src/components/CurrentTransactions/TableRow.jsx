import React, {Fragment} from 'react';
import PropTypes from 'prop-types';

const TableRow = ({
    index,
    id,
    origin,
    destination,
    startSendTimestamp,
    endReceiveTimestamp,
    error,
    completed,
    amount
}) => {
    const classStyle = error ? 'table-danger' : '';
    return (
        <tr className={classStyle}>
            <th scope='row'>{index}</th>
            <td>{id}</td>
            <td>{origin.nodeLocation}</td>
            <td>{destination.nodeLocation}</td>
            <td>{amount} nano</td>
            { completed ?
                <Fragment>
                { error ?
                    <Fragment>
                        <td>ERROR</td>
                        <td>ERROR OCCURRED</td>
                    </Fragment> : <Fragment>
                        <td>{(endReceiveTimestamp - startSendTimestamp)/1000} Seconds</td>
                        <td>Completed</td>
                    </Fragment>
                }
                </Fragment>
                : <Fragment>
                    <td>Pending...</td>
                    <td>Pending...</td>
                </Fragment>
            }
        </tr>
    );
};

TableRow.propTypes = {
    index: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    origin: PropTypes.object.isRequired,
    destination: PropTypes.object.isRequired,
    time: PropTypes.number,
    amount: PropTypes.number,
    completed: PropTypes.bool,
    endReceiveTimestamp: PropTypes.number,
    startSendTimestamp: PropTypes.number
};

export default TableRow;