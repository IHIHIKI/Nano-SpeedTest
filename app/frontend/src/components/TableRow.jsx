import React from 'react';
import PropTypes from 'prop-types';

const TableRow = ({index, hash, origin, destination, time, amount, completed}) => {
    return (
        <tr>
            <th scope="row">{index}</th>
            <td>{hash}</td>
            <td>{origin}</td>
            <td>{destination}</td>
            <td>{time} Seconds</td>
            <td>{amount}</td>
            <td>{completed}</td>
        </tr>
    );
};

TableRow.propTypes = {
    index: PropTypes.number.isRequired,
    hash: PropTypes.string.isRequired,
    origin: PropTypes.string.isRequired,
    destination: PropTypes.string.isRequired,
    time: PropTypes.number,
    amount: PropTypes.number,
    completed: PropTypes.string
};

export default TableRow;