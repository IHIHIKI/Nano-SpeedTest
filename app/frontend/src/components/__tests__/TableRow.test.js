import React from 'react';
import renderer from 'react-test-renderer';
import TableRow from '../CurrentTransactions/TableRow';
import { shallow } from 'enzyme';

const props = {
    index: 0,
    id: 123,
    origin: {},
    destination: {},
    transactionHashSending: 'ABC123',
    transactionHashReceiving: 'ZXY321'
};

it('shallow renders without crashing', () => {
    shallow(<TableRow  {...props}/>);
});

it(`renders snapshot correctly`, () => {
    const tree = renderer.create(
        <TableRow {...props}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
