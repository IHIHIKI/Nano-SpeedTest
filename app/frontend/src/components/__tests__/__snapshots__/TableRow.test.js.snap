// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<tr
  className=""
>
  <td />
  <td />
  <td>
     nano
  </td>
  <td>
    Pending...
  </td>
  <td>
    Pending...
  </td>
  <td>
    Pending...
  </td>
  <td>
    Pending...
  </td>
</tr>
`;
